src/handlers/bot_pull_requests.rs

```diff
@@ -0,0 +1,66 @@
+use indexmap::IndexSet;
+use std::sync::atomic::AtomicBool;
+use std::sync::{LazyLock, Mutex};
+
+use crate::github::{IssueRepository, IssuesAction, PrStatus};
+use crate::{github::Event, handlers::Context};
+
+pub(crate) async fn handle(ctx: &Context, event: &Event) -> anyhow::Result<()> {
+    let Event::Issue(event) = event else {
+        return Ok(());
+    };
+    if event.action != IssuesAction::Opened {
+        return Ok(());
+    }
+    if !event.issue.is_pr() {
+        return Ok(());
+    }
+
+    // avoid acting on our own open events, otherwise we'll infinitely loop
+    if event.sender.login == ctx.username {
```

Comment:
I think that when the PR is reopened, it will trigger the "reopened" event, and not the "opened" event (https://docs.github.com/en/rest/using-the-rest-api/issue-event-types?apiVersion=2022-11-28#reopened). But it doesn't hurt to be extra safe here.